common/achievement/bucket.ts

```diff
@@ -1,14 +1,23 @@
 import moment from 'moment';
-import { BucketFormula, DefaultBucket, GenericBucketConfig, TimeBucket, ContextMatch, ContextSubcourse } from './types';
+import { BucketFormula, DefaultBucket, GenericBucketConfig, TimeBucket, ContextMatch, ContextSubcourse, ContextLecture } from './types';
 
 type BucketCreatorDefs = Record<string, BucketFormula>;
 
 function createLectureBuckets<T extends ContextMatch | ContextSubcourse>(data: T): TimeBucket[] {
     if (!data.lecture || data.lecture.length === 0) {
         return [];
     }
+    data.lecture.sort((a, b) => a.start.getTime() - b.start.getTime());
+    const filteredLectures: ContextLecture[] = data.lecture.filter((lecture, index, array) => {
+        if (index === 0) {
+            return true;
+        }
+        const previousEndTime = new Date(array[index - 1].start.getTime() + array[index - 1].duration * 60000);
+        return lecture.start >= previousEndTime;
+    });
```

Comment:
This function is taking out lectures that overlap in a unwanted way. Having 2 lectures with the same relation would result in 2 buckets taking the same events, therefore rewarding the user with a value of 2 instead of 1.

I had a talk with AK and we came up with a different solution, perhibitting the creation of 2 lectures, overlapping in their timeframe.

Whats your opinion?

Reply:
👍 I like the new idea.

Reply:
https://github.com/corona-school/project-user/issues/1108